recipes/openldap/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+add_executable(${PROJECT_NAME} test_package.cpp)
+# For gcc command, the -lA must come before -lB for static libs, if A depends on B.
+list(REMOVE_ITEM CONAN_LIBS "ldap")
+list(INSERT CONAN_LIBS 0 "ldap")
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```

```suggestion
conan_basic_setup(TARGETS)

find_package(openldap CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} openldap::openldap)
```

As I suggested on https://github.com/sophieeihpos/conan-center-index/pull/1, this linker order must be solved by the cmake file generated by Conan